packages/lib/src/vis/rgb/utils.ts

```diff
@@ -33,12 +34,14 @@ export function toRgbSafeNdArray(
   return toTypedNdArray(ndArr, Uint8Array);
 }
 
-export function flipLastDimension<T extends TypedArray>(
-  dataArray: NdArray<T>
-): NdArray<T> {
-  const { shape } = dataArray;
-  const steps = shape.map((_, index) => (index === shape.length - 1 ? -1 : 1));
+export function getDataTexture3D(
+  values: NdArray<Uint8Array | Uint8ClampedArray | Float32Array>
+): DataTexture3D {
+  const { rows, cols } = getDims(values);
 
-  const flippedView = dataArray.step(...steps);
-  return createArrayFromView(flippedView);
+  const texture = new DataTexture3D(values.data, 3, cols, rows);
```

Comment:
Let me sum up so I get this right. `DataTexture3D` args are supposed to be `data, width, height, depth`. Semantically, we should then put `values.data, cols, rows, 3` in this order.

_But_, you put `3` first to be able to recover the RGB channels in the shader by using:
```
        float red = texture(data, vec3(0., coords)).r;
        float green = texture(data, vec3(0.5, coords)).r;
        float blue = texture(data, vec3(1., coords)).r;
```
?

Reply:
I tried `texture(data, vec3(coords, 0.)).r` at first, with `3` passed as `depth` , but was getting the wrong colors. I guess, we already pass `cols, rows` (i.e. the ndarray's reversed shape) for `width, height`, so it kinda makes sense?